test/Jamfile.v2

```diff
@@ -81,7 +82,22 @@ project
                 : <link>static <variant>release <library>static_refcounting_plugin $(RDYNAMIC) <test-info>always_show_run_output
         ]
         [ run ../example/tutorial9/tutorial9.cpp ]
+		# test for shared libraries
         [ compile-fail section_name_too_big.cpp ]
-        [ run shared_library_concurrent_load_test.cpp /boost/thread//boost_thread : : library1 library2 my_plugin_aggregator refcounting_plugin : <link>shared ]
-    ;
+		[ run shared_library_concurrent_load_test.cpp /boost/thread//boost_thread : : library1 library2 my_plugin_aggregator refcounting_plugin : <link>shared ]
+     ;
+	 
+	.argv = [ modules.peek : ARGV ] ;
+	if ! --no-smart-lib in $(.argv) 
```

Comment:
Please, replace tabs with spaces.

Also, note that majority of tests are run without any additional flags [here](http://www.boost.org/development/tests/develop/developer/dll.html). So `--no-smart-lib` is not usable. Just make the tests to always run, and in source files of tests add `#if __cplusplus > 201402L .... #else int main() {} #endif`